eng/scripts/build.ps1

```diff
@@ -1,67 +1,74 @@
 #Requires -Version 7.0
-param($filter, [switch]$clean, [switch]$vet, [switch]$generate, [switch]$skipBuild)
+param([string]$filter, [switch]$clean, [switch]$vet, [switch]$generate, [switch]$skipBuild, [string]$config = "autorest.md", [string]$outputFolder)
 
 . $PSScriptRoot/meta_generation.ps1
+. $PSScriptRoot/get_module_dirs.ps1
 
-$startingDirectory = Get-Location
-$root = Resolve-Path ($PSScriptRoot + "/../..")
-Set-Location $root
-$sdks = @{};
-
-foreach ($sdk in (./eng/scripts/get_module_dirs.ps1 -serviceDir 'sdk/...')) {
-    $name = $sdk | split-path -leaf
-    $sdks[$name] = @{
-        'path'      = $sdk;
-        'clean'     = $clean;
-        'vet'       = $vet;
-        'generate'  = $generate;
-        'skipBuild' = $skipBuild;
-        'root'      = $root;
-    }
-}
 
-$keys = $sdks.Keys | Sort-Object;
-if (![string]::IsNullOrWhiteSpace($filter)) {
-    Write-Host "Using filter: $filter"
-    $keys = $keys.Where( { $_ -match $filter })
-}
-
-$keys | ForEach-Object { $sdks[$_] } | ForEach-Object {
-    Push-Location $_.path
-
-    if ($_.clean) {
-        Write-Host "##[command]Executing go clean -v ./... in " $_.path
+function Process-Sdk ($path) {
+    if ($clean) {
+        Write-Host "##[command]Executing go clean -v ./... in " $path
         go clean -v ./...
     }
 
-    if ($_.generate) {
-        Write-Host "##[command]Executing autorest.go in " $_.path
-        $autorestPath = $_.path + "/autorest.md"
+    if ($generate) {
+        Write-Host "##[command]Executing autorest.go in " $path
+        $autorestPath = $path + "/" + $config
 
         if (ShouldGenerate-AutorestConfig $autorestPath) {
             Generate-AutorestConfig $autorestPath
             $removeAutorestFile = $true
         }
 
         $autorestVersion = "@autorest/go@4.0.0-preview.23"
```

Comment:
Should we look the version up from a file or pass via argument?  This string is likely to get out of date

Reply:
My idea was that this was the single place we'd bump the autorest version. 

I could move it out to another file, but in any case there will be one place to rule them all.